index.bs

```diff
@@ -661,31 +661,31 @@ option. If <code><a for="underlying source">type</a></code> is set to <code>unde
 </div>
 
 <h5 id="rs-pipe-through" method for="ReadableStream" lt="pipeThrough(transform, options)">pipeThrough({
-<var ignore>writable</var>, <var ignore>readable</var> }, <var ignore>options</var>)</h5>
+<var ignore>writable</var>, <var ignore>readable</var> }, { <var>preventClose</var>, <var>preventAbort</var>,
+<var>preventCancel</var>, <var>signal</var> } = {})</h5>
 
 <div class="note">
   The <code>pipeThrough</code> method provides a convenient, chainable way of <a>piping</a> this <a>readable stream</a>
   through a <a>transform stream</a> (or any other <code>{ writable, readable }</code> pair). It simply pipes the stream
   into the writable side of the supplied pair, and returns the readable side for further use.
 
-  Piping a stream will generally <a lt="locked to a reader">lock</a> it for the duration of the pipe, preventing any
-  other consumer from acquiring a reader.
-
-  This method is intentionally generic; it does not require that its <emu-val>this</emu-val> value be a
-  {{ReadableStream}} object. It also does not require that its <code>writable</code> argument be a {{WritableStream}}
-  instance, or that its <code>readable</code> argument be a {{ReadableStream}} instance.
+  Piping a stream will <a lt="locked to a reader">lock</a> it for the duration of the pipe, preventing any other
+  consumer from acquiring a reader.
 </div>
 
 <emu-alg>
-  1. If _writable_ is *undefined* or _readable_ is *undefined*, throw a *TypeError* exception indicating that undefined
-     values are not permitted.
-     <p class="note"><emu-val>undefined</emu-val> is special-cased here to make it easier to debug errors such as
-     <code>rs.pipeThrough(writable, readable)</code> which otherwise would silently do nothing. <code><a
-     lt="pipeThrough()" for="ReadableStream">pipeThrough()</a></code> is
-     otherwise agnostic about the values that it passes through.</p>
-  1. Let _promise_ be ? Invoke(*this*, `"pipeTo"`, « _writable_, _options_ »).
-  1. If Type(_promise_) is Object and _promise_ has a [[PromiseIsHandled]] internal slot, set
-     _promise_.[[PromiseIsHandled]] to *true*.
+  1. If ! IsReadableStream(*this*) is *false*, throw a *TypeError* exception.
+  1. If ! IsWritableStream(_writable_) is *false*, throw a *TypeError* exception.
+  1. If ! IsReadableStream(_readable_) is *false*, throw a *TypeError* exception
+  1. Set _preventClose_ to ! ToBoolean(_preventClose_), set _preventAbort_ to ! ToBoolean(_preventAbort_), and set
+     _preventCancel_ to ! ToBoolean(_preventCancel_).
+  1. If _signal_ is not *undefined*, and _signal_ is not an instance of the `<a idl>AbortSignal</a>` interface, throw a
+     *TypeError* exception.
+  1. If ! IsReadableStreamLocked(*this*) is *true*, throw a *TypeError* exception.
```

Comment:
Interesting that these now throw sync exceptions instead of returning a rejected promise that gets ignored. This is probably the right thing to do...

Reply:
The alternative would be to error `readable`. That would have the benefit that the whole pipe chain would be errored in the normal case, and so you'd get a rejection rather than an exception. But it wouldn't work properly if `readable` was locked. It also would be a big departure from our behaviour up until now of passing through `readable` unchanged.

On balance, I think throwing an exception here is the best approach.